src/pke/lib/scheme/ckksrns/ckksrns-leveledshe.cpp

```diff
@@ -360,15 +360,52 @@ std::vector<DCRTPoly::Integer> LeveledSHECKKSRNS::GetElementForEvalAddOrSub(
   const auto cryptoParams =
       std::static_pointer_cast<CryptoParametersCKKSRNS>(
           ciphertext->GetCryptoParameters());
-  double scFactor =
-      cryptoParams->GetScalingFactorReal(ciphertext->GetLevel());
 
-  DCRTPoly::Integer intScFactor = static_cast<uint64_t>(scFactor + 0.5);
+  double scFactor;
```

Comment:
never leave uninitialized variables

Reply:
It's initialized in the if-else loop below. Should I do:
double scFactor = cryptoParams->GetScalingFactorReal(ciphertext->GetLevel());
if (cryptoParams->GetRescalingTechnique() == FLEXIBLEAUTOEXT && ciphertext->GetLevel() == 0) {
    scFactor = cryptoParams->GetScalingFactorRealBig(ciphertext->GetLevel());
 }

Reply:
no, it would be difficult to read. "double scFactor = 0;" should be sufficient.

Reply:
Updated to the above